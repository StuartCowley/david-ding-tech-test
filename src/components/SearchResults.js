import React from "react";
import "../styles/search-results.css";
const SearchResults = ({ results }) => {
  if (!results.length) {
    return <p>No results</p>;
  } else {
    return (
      <>
        <p>Search Results</p>
        {results.map((result, index) => (
          <img
            key={index}
            className="card-image"
            alt="moon-pic"
            src={result}
          ></img>
        ))}
      </>
    );
  }
};

export default SearchResults;
