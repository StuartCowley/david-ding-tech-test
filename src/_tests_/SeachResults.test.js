import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  it("renders correctly", () => {
    const { asFragment } = render();
    <SearchResults />;

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct text", () => {
    render(<SearchResults />);
    expect(screen.getByText("Search Results")).toBeInTheDocument();
  });
});
