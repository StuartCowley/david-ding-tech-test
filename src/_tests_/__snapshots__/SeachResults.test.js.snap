// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <p
    class="no-results-msg"
  >
    No results!
  </p>
</DocumentFragment>
`;
